src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -311,6 +311,61 @@ g.test('limit,better_than_supported')
     t.shouldReject('OperationError', adapter.requestDevice({ requiredLimits }));
   });
 
+g.test('limit,out_of_range')
+  .desc(
+    `
+    Test that specifying limits that are out of range (<0, >MAX_SAFE_INTEGER, >2**31-2 for 32-bit
+    limits, =0 for alignment limits) produce the appropriate error (TypeError or OperationError).
+    `
+  )
+  .params(u =>
+    u
+      .combine('limit', kLimits)
+      .beginSubcases()
+      .expand('value', function* () {
+        yield -1;
+        yield 0;
+        yield 2 ** 32 - 2;
+        yield 2 ** 32 - 1;
+        yield 2 ** 32;
+        yield 2 ** 32 + 1;
+        yield 2 ** 32 + 2;
+        yield Number.MAX_SAFE_INTEGER - 1;
+        yield Number.MAX_SAFE_INTEGER;
+        yield Number.MAX_SAFE_INTEGER + 2;
+        yield 2 ** 64 - 1;
+        yield 2 ** 64;
+        yield Number.MAX_VALUE;
+      })
+  )
+  .fn(async t => {
+    const { limit, value } = t.params;
+
+    const gpu = getGPU(t.rec);
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+    const limitInfo = getDefaultLimitsForAdapter(adapter)[limit];
+
+    const requiredLimits = {
+      [limit]: value,
+    };
+
+    const errorName =
+      value < 0 || value > Number.MAX_SAFE_INTEGER
+        ? 'TypeError'
+        : limitInfo.class === 'maximum' && value > adapter.limits[limit]
+        ? 'OperationError'
+        : limitInfo.class === 'alignment' && (value > 2 ** 31 || !isPowerOfTwo(value))
```

Comment:
The spec does not actually currently say that `2**32` or `2**53` is an error for alignment limits. I'm proposing it to the spec:
https://github.com/gpuweb/gpuweb/pull/4358